### DISCUSSION TITLE
Can I run this on a Photos library with Optimse Mac Storage enabled?

### DISCUSSION THREAD

**Discussion by ilium007** [RhetTbull/osxphotos#1426] [General]
I have Optimse Mac Storage enabled given the low storage space on my Macbook Air. I want to export my library to an external SSD. Will osxphotos download the large version from iCloud upon export?

![image](https://github.com/RhetTbull/osxphotos/assets/732766/7b786ac8-4325-4f30-a2ec-88a211542556)

**Comment by RhetTbull** [comment]
Yes, if you use `--download-missing`, osxphotos will download the original from iCloud and export it. This will obviously be slow as each asset needs to be downloaded. (osxphotos actually tells Photos to download it and then export it then Photos will delete it to conserve space). I would start with `--download-missing --use-photokit` which tells Photos to use a different interface that's faster. This doesn't work for everyone though so if that gives you any errors, drop `--use-photokit` and try again. 

If you just want an offline copy of your photos and don't need specific features form osxphotos, you might also consider [iCloud Photos Downloader](https://github.com/icloud-photos-downloader/icloud_photos_downloader) which downloads directly from iCloud. I do not use this and am not affiliated with the project but it's an option that might work in your case.

**Comment by ilium007** [comment]
Thanks - I need the date export so I will give this one a go!
